src/Whim.FloatingWindow/FloatingUtils.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Immutable;
+
+namespace Whim.FloatingWindow;
+
+/// <summary>
+/// Describe the result of the UpdateWindowRectangle
+/// </summary>
+internal enum UpdateWindowStatus
+{
+	/// <summary>
+	/// Could not obtain the rectangle for the window
+	/// </summary>
+	Error,
+
+	/// <summary>
+	/// Nothing was changed.
+	/// </summary>
+	NoChange,
+
+	/// <summary>
+	/// The new position was calculated, and the dict updated.
+	/// </summary>
+	Updated,
+}
+
+/// <summary>
+/// Provide methods for the floating engines
+/// </summary>
+internal static class FloatingUtils
+{
+	/// <summary>
+	/// Update the position of a <paramref name="window"/> from the <paramref name="dict"/>.
+	/// </summary>
+	/// <param name="context"></param>
+	/// <param name="dict"></param>
+	/// <param name="window"></param>
+	/// <returns>A tuple with the maybe updated <paramref name="dict"/>, and its <param cref="UpdateWindowStatus"></param></returns>
+	public static (
```

Comment:
I think it's possible to remove the `enum` by switching to returning `ImmutableDictionary<>` and:

- returning `null` for an error
- returning the provided `dict` argument if there isn't a change
- returning the `newDict` if there is a change

The handling code can do a comparison of the `dict` passed in and the result of `UpdateWindowRectangle`. 

I think this approach is overall an idiomatic way of handling immutable data structures.

Reply:
Done